arangod/Aql/OptimizerRules.cpp

```diff
@@ -4703,7 +4709,9 @@ void arangodb::aql::distributeSortToClusterRule(
   bool modified = false;
 
   for (auto& n : nodes) {
-    auto const& remoteNodeList = n->getDependencies();
+    // intentional copy of the dependencies, as we will be modifying 
+    // dependencies later on
+    auto const remoteNodeList = n->getDependencies();
     auto gatherNode = ExecutionNode::castTo<GatherNode*>(n);
     TRI_ASSERT(remoteNodeList.size() > 0);
     auto rn = remoteNodeList[0];
```

Comment:
Is this copy really necessary here? As far as I can see remoteNodeList is only used these 4 lines.